### ISSUE TITLE
GM Notes Header Button

### ISSUE THREAD

**Issue by mouse0270** [bithir/gm-notes#27]
Is there any particular reason you append the header button in the `renderXYZ` hook instead of registering it as a header button in the `getXYZHeaderButtons` hooks?

**Comment by bithir** [comment]
No, no reason - it is was just made this way when I took over the project.

**Comment by mouse0270** [comment]
If I make a pull request can we update it?

**Comment by bithir** [comment]
Of course, I happily accept pull request.  Can you detail what the advantages are?

**Comment by mouse0270** [comment]
Using the getHeadersButtons hook is the *Standard Foundry API* method used for registering buttons in the header. This method also makes it so other modules who are getting and interaction with header buttons knows the button exists.

For example, my module WindowTabs, groups windows together and only shows buttons for the currently active tab. So if the button hasn't been registered as a header button, my module has no way to know it exists.

Basically, its the proper way to do it, and should only be circumvented if absolutely needed.

**Comment by bithir** [comment]
I've been looking at the hook - but I can't see how I change the indicator on the button upon changes.
I must be missing something - I can't close and open the app to just update the header button.

So I would still need to listen to the render hook to do the update?

**Comment by mouse0270** [comment]
Sorry I dont use this module, could you tell me what is changing about the button on the render action?

**Comment by bithir** [comment]
The icon (and text) changes if you are looking at an item/actor/light/tile depending if they have a note or not.
Example: No note
![image](https://github.com/bithir/gm-notes/assets/19734037/8297ea2b-3cba-4b33-8ab7-dac6c95e307a)
Example: Have note
![image](https://github.com/bithir/gm-notes/assets/19734037/2c50c799-86a5-4d34-801e-9cff0004792d)

**Comment by bithir** [comment]
I'm suspecting that this is the reason other modules are also doing it in the render hook.  If the button depends on the underlying data, then using the button header seems quite involved, compared to just adding it in the render hook.

**Comment by mouse0270** [comment]
Well, changing the icon, is easily enough, but the color is a littler tricker, but I can do it via css. I try and work on it later.

Impossible probably not, being tricky, probably so. haha

**Comment by bithir** [comment]
Could add a dummy button in the button hook and replace it in the render?

**Comment by bithir** [comment]
The big issue is that the header button hook only triggers once (when the sheet is opened), so there is no possible way to change the icon/text if the underlying data changes.

I have another module which has a header button but this one does not change, where I will move over to use the button header hook.

**Comment by mouse0270** [comment]
Made the pull request, which because also happens to appear to also add support for #1 because I am attaching to any application that has a `.document`

**Comment by mouse0270** [comment]
You should be able to close this issue and #1

**Comment by bithir** [comment]
Resolved by #29
